server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1589,6 +1509,148 @@ static String resolveExpression(String expression, LongSupplier getTime) {
         }
     }
 
+    public static final class ExplicitResourceNameFilter {
+
+        private ExplicitResourceNameFilter() {
+            // Utility class
+        }
+
+        public static List<String> filter(Context context, List<String> expressions) {
+            List<String> result = new ArrayList<>(expressions.size());
+            for (ExpressionIterable.Expression expression : new ExpressionIterable(context, expressions)) {
+                validateAliasOrIndex(expression);
+                if (expression.isWildcard() || expression.isExclusion() || ensureAliasOrIndexExists(context, expression.toString())) {
+                    result.add(expression.isExclusion() ? "-" + expression : expression.toString());
+                }
+            }
+            return result;
+        }
+
+        @Nullable
+        private static boolean ensureAliasOrIndexExists(Context context, String expression) {
```

Comment:
This is basically unchanged, but now it might return `false` rather than throw a "not found" exception, in order to make it reusable in the case that the `ignore_unavailable` request option is `true`.